src/main/kotlin/org/jetbrains/intellij/platform/gradle/argumentProviders/SplitModeArgumentProvider.kt

```diff
@@ -0,0 +1,19 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+
+package org.jetbrains.intellij.platform.gradle.argumentProviders
+
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.process.CommandLineArgumentProvider
+
+/**
+ * Provides command line arguments for enabling Split Mode.
+ */
+class SplitModeArgumentProvider(
+    @Input val splitMode: Provider<Boolean>,
+) : CommandLineArgumentProvider {
+
+    override fun asArguments() = listOfNotNull(
+        "splitMode".takeIf { splitMode.get() },
```

Comment:
This argument must be the first one. Are we sure that other providers won't execute before this one?

Reply:
Thanks for highlighting that. I covered that scenario with https://github.com/JetBrains/intellij-platform-gradle-plugin/commit/38fd1c359d42b98c0c43d8537b139b1fd4a4154c#diff-053331baad74a29668f8bf6d9110d44e2f8a13d3118d39599eeaf13ec67654d1R63-R65